### PR TITLE
New implementation process execution.

### PR DESCRIPTION
**Pull request by svandenhoek**
Implemented a new process execution which allows for directly pipe-ing data between processes and allows writing process output directly to the Mapper/Reducer context without first storing it in a variable. Additional updates to other files (such as the READMEs and scripts) have been added accordingly as well. The old code is currently moved to a package called "legacy".

### CODE REVIEW

File: hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/processes/SamRecordSink.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	 * implementation!
+	 * 
+	 * @param item
+	 *            {@link SAMRecord}
+	 */
+	@Override
+	protected void digestStreamItem(SAMRecord item)
+	{
+	}
+
```
Comment (erwinwinder) [inline hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/processes/SamRecordSink.java]: Make digestStreamItem abstract, so subclasses are required to override it

File: hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/processes/StringSink.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+	/**
+	 * Digests a single line from the {@link InputStream}. Be sure to create a custom {@code @Override} implementation!
+	 * 
+	 * @param item
+	 *            {@link String}
+	 */
+	@Override
+	protected void digestStreamItem(String item)
+	{
```
Comment (erwinwinder) [inline hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/processes/StringSink.java]: Make abstract

File: hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/processes/SamRecordSink.java (hunk 1 above)
Comment (svandenhoek) [inline hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/processes/SamRecordSink.java]: Added it for the Javadoc, though forgot to add abstract. Will adjust right away.

File: hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/processes/StringSink.java (hunk 2 above)
Comment (svandenhoek) [inline hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/processes/StringSink.java]: Added it for the Javadoc, though forgot to add abstract. Will adjust right away.

**Pull request merged**